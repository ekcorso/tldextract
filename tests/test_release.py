"""Test the library maintainer release script."""

import pytest

from scripts import release


def test_happy_path() -> None:
    """Test the release script happy path."""
    assert release
    pytest.xfail("Not implemented yet")
