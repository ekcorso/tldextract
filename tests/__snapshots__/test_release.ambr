# serializer version: 1
# name: test_happy_path
  _CallList([
    _Call(
      tuple(
        'dist',
      ),
      dict({
      }),
    ),
  ])
# ---
# name: test_happy_path.1
  _CallList([
    _Call(
      tuple(
        'https://api.github.com/repos/ekcorso/releasetestrepo2/releases/generate-notes',
      ),
      dict({
        'headers': dict({
          'Accept': 'application/vnd.github+json',
          'Authorization': 'Bearer fake-token',
          'X-GitHub-Api-Version': '2022-11-28',
        }),
        'json': dict({
          'tag_name': '5.0.1',
        }),
      }),
    ),
    _Call(
      tuple(
        'https://api.github.com/repos/ekcorso/releasetestrepo2/releases',
      ),
      dict({
        'headers': dict({
          'Accept': 'application/vnd.github+json',
          'Authorization': 'Bearer fake-token',
          'X-GitHub-Api-Version': '2022-11-28',
        }),
        'json': dict({
          'body': '''
            * Bugfixes
                * Indicate MD5 not used in a security context (FIPS compliance) ([#309](https://github.com/john-kurkowski/tldextract/issues/309))
            * Misc.
                * Increase typecheck aggression
            
            **Full Changelog**: fake-body
          ''',
          'draft': True,
          'name': '5.0.1',
          'prerelease': False,
          'tag_name': '5.0.1',
        }),
      }),
    ),
  ])
# ---
# name: test_happy_path.2
  _CallList([
    _Call(
      tuple(
        list([
          'git',
          'tag',
          '-a',
          '5.0.1',
          '-m',
          '5.0.1',
        ]),
      ),
      dict({
        'check': True,
      }),
    ),
    _Call(
      tuple(
        list([
          'rm',
          '-rf',
          PosixPath('dist'),
        ]),
      ),
      dict({
        'check': True,
      }),
    ),
    _Call(
      tuple(
        list([
          'python',
          '-m',
          'build',
        ]),
      ),
      dict({
        'check': True,
      }),
    ),
    _Call(
      tuple(
        list([
          'ls',
          '-l',
          PosixPath('dist'),
        ]),
      ),
      dict({
        'check': True,
      }),
    ),
    _Call(
      tuple(
        list([
          'twine',
          'upload',
          '--repository',
          'testpypi',
          PosixPath('dist/*'),
        ]),
      ),
      dict({
        'check': True,
      }),
    ),
    _Call(
      tuple(
        list([
          'git',
          'push',
          '--tags',
          'origin',
          'master',
        ]),
      ),
      dict({
        'check': True,
      }),
    ),
  ])
# ---
# name: test_happy_path.3
  '''
  Starting the release process...
  Checking for github token environment variable...
  GITHUB_TOKEN environment variable is good to go.
  Version 5.0.1 tag added successfully.
  Previous dist folder removed successfully.
  Build created successfully.
  Contents of dist folder:
  Contents of tar files in dist folder:
  Build verified successfully.
  Release created successfully: https://github.com/path/to/release
  
  '''
# ---
# name: test_happy_path.4
  '''
  WARNING: dist folder contains incorrect number of files.
  
  '''
# ---
